### DISCUSSION TITLE
Evaluating rate of convergence

### DISCUSSION THREAD

**Discussion by jagpdev** [ContinualAI/avalanche#1071] [Q&A] [answered]
Hi Everyone,

I'm researching how continual learning methods can help improve the speed at which we converge to a solution. The idea is that training a model on Task A and using the same model to train Task B could result in Task B converging to a solution quicker by reusing the parameters from Task A. In order to test this, I've trained a model using the Joint Strategy on Task B i.e. training a model from scratch comparing this to a Naive strategy where I'd like the first experience to train the model on Task A and then the second experience on Task B.

For the Naive strategy I've set up the following, task B in this case is training the model to recognize classes 2 and 3:

```
train_set = CIFAR10(default_dataset_location('cifar10'), train=True, download=True)
test_set = CIFAR10(default_dataset_location('cifar10'), train=False, download=True)

#Filter the dataset so we fetch classes 0,1,2,3 only
train_set, test_set = filterClasses(train_set, test_set, classes=[0,1,2,3])

benchmark = nc_benchmark(
             train_dataset=train_set,
             test_dataset=test_set,
             n_experiences=2,
             fixed_class_order=[0,1,2,3],
             task_labels=True,
             train_transform=_default_cifar10_train_transform,
             eval_transform=_default_cifar10_eval_transform)

#Number of classes is 2 as we'd like to train the model on classes 0 and 1, then retrain the model on 2 and 3.
model = SimpleCNN(num_classes=2)


eval_plugin = EvaluationPlugin(
    accuracy_metrics(experience=True),
    loggers=[InteractiveLogger()],
    strict_checks=False
)


cl_strategy = Naive(
    model, SGD(model.parameters(), lr=0.01, momentum=0.9),
    CrossEntropyLoss(), train_mb_size=1000, train_epochs=1, eval_mb_size=200,
    evaluator=eval_plugin)


print('Starting experiment...')
for experience in benchmark.train_stream:
    experience.dataset.targets  = experience.dataset.targets - np.min(experience.dataset.targets)
    res = cl_strategy.train(experience)
    print('Training completed')
```

The problem is this line:
experience.dataset.targets  = experience.dataset.targets - np.min(experience.dataset.targets)

This line bounds the targets to 0,1 for the second experience instead of 2,3 because the model only expects 2 classes, so targets>2 results in an index error.
Is this the best way to go about this problem?

**Comment by AntonioCarta** [comment, marked as answer]
You can remove that line since there is no need to change the targets. You can either use an `IncrementalClassifier`, which automatically expands the head with new classes, or use a `MultiHeadClassifier`.

**Reply by jagpdev** [reply]
Thanks Antonio, MultiHeadClassifier was what I was looking for!
